### PR TITLE
Mpuncel/ds contention

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: pkg/ds/farm.go [hunk 1]
```diff
@@ -502,83 +504,73 @@ func (dsf *Farm) releaseLock(unlocker consulutil.Unlocker) {
...
+
+	allDaemonSets, err := dsStore.List()
+	if err != nil {
+		return ds_fields.DaemonSet{}, false, util.Errorf("listing daemon sets failed so could not check for contention: %s", err)
 	}
 
 	// If this daemon set has a node selector set to Everything, check the labels
 	// of other daemon sets
-	for _, child := range dsf.children {
+	for _, otherDS := range allDaemonSets {
```
Comment (rudle) [inline pkg/ds/farm.go]: d'oh

File: bin/p2-dsctl/main.go [hunk 2]
```diff
@@ -129,7 +131,14 @@ func main() {
 			log.Fatalf("Error occurred: %v", err)
 		}
 
-		ds, err := dsstore.Create(manifest, minHealth, name, selector, podID, *createTimeout)
+		ctx, cancelFunc := transaction.New(context.Background())
+		defer cancelFunc()
+		ds, err := dsstore.Create(ctx, manifest, minHealth, name, selector, podID, *createTimeout)
```
Comment (rudle) [inline bin/p2-dsctl/main.go]: Is this for the future? What's the advantage of a transaction with only one operation?

Comment (mpuncel) [inline bin/p2-dsctl/main.go]: right now the advantage is the deferred creation of the daemon set. `Create()` creates a `DeamonSet` which is required to pass to `DSContends`, but the problem is i don't want the creation to succeed until i know there isn't contention.

I agree that there are ways to accomplish this without using a transaction but i figured we probably want this to be transactional soon anyway, i think audit log records for daemon set creations will be nice

**Comment by mpuncel** [comment]
added a newline to the end of the "checking for contention" message

**Pull request merged**